### ISSUE TITLE
Bug Report: error when running `apt update` command

### ISSUE THREAD

**Issue by pkhadka56** [CircleCI-Public/cimg-gcp#27]
**Describe the bug**
Got the following error when running `apt update` 
image: **cimg/gcp:2023.12**
```
E: The repository 'https://apt.kubernetes.io kubernetes-xenial Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
W: https://download.docker.com/linux/ubuntu/dists/jammy/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
W: GPG error: https://packages.cloud.google.com/apt cloud-sdk InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY C0BA5CE6DC6315A3
E: The repository 'https://packages.cloud.google.com/apt cloud-sdk InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.

Exited with code exit status 100
```

**To Reproduce**
Use the image and run `sudo apt update -y`

**Expected behavior**
Command runs successfully

**Screenshots and Build Links**
<img width="1043" alt="image" src="https://github.com/CircleCI-Public/cimg-gcp/assets/8862495/628fa4cd-a039-4fd1-99df-e43c4e10d4e3">

**Comment by peterwwillis** [comment]
Could be related to https://github.com/kubernetes/release/issues/3485

> [As announced previously](https://kubernetes.io/blog/2023/08/31/legacy-package-repository-deprecation/), the legacy Kubernetes package repositories (apt.kubernetes.io and yum.kubernetes.io, also known as packages.cloud.google.com) have been shut down (removed) as of 2024-03-04.

**Comment by vironeen** [comment]
It appears that adding this command before the apt commands has resolved the packages.cloud.google.com error for me.  
```
curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo gpg --dearmor --yes -o /etc/apt/trusted.gpg.d/cloud.google.gpg
```
Note this is the gpg file specified in the gcloud docs.
https://cloud.google.com/sdk/docs/install-sdk#deb

**Comment by peterwwillis** [comment]
For those running into this, appears a new container has been issued with a fix: `cimg/gcp:2024.03.1`

**Comment by mahadzar81** [comment]
This appear same issue to me recently
Below command works for me

`curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo gpg --dearmor --yes -o /etc/apt/trusted.gpg.d/cloud.google.gpg`
